### PR TITLE
Refactor grace note tests with factory

### PR DESCRIPTION
**Pull request by Silverwolf90**
Added `GraceNote` and `GraceNoteGroup` to the `Factory`.

### CODE REVIEW

File: src/factory.js [hunk 1]
```diff
@@ -118,6 +121,19 @@ export class Factory {
...
 
+  GraceNote(noteStruct) {
+    const note = new GraceNote(noteStruct);
+    if (this.stave) note.setStave(this.stave);
+    note.setContext(this.context);
+    return note;
+  }
+
+  GraceNoteGroup(params) {
+    const group = new GraceNoteGroup(params.notes, params.slur);
```
Comment (Silverwolf90) [inline src/factory.js]: should this method have a `params.options` (for consistency with the other factory methods) even though the `GraceNoteGroup` constructor's second argument is a boolean?

File: src/factory.js [hunk 2]
```diff
@@ -173,6 +189,19 @@ export class Factory {
...
 
+  Beam(params) {
+    params = setDefaults(params, {
+      notes: [],
+      options: {
+        autoStem: false,
+      },
+    });
+
+    const beam = new Beam(params.notes, params.options.autoStem).setContext(this.context);
```
Comment (Silverwolf90) [inline src/factory.js]: (Note that this code is also duplicated in  https://github.com/0xfe/vexflow/pull/417 )

The inverse of the other comment, should the `params.options` exist here even though beam doesn't take an `options` object in the constructor?

File: src/factory.js (hunk 1 above)
Comment (0xfe) [inline src/factory.js]: If it's unused, you can leave it for now. It's easy to add it if we need it without breaking users.

File: src/factory.js (hunk 2 above)
Comment (0xfe) [inline src/factory.js]: Yes, that is fine. This is actually a good case for it, and a way to break out of long unnamed parameter lists.

**Comment by 0xfe** [comment]
Looks good. Merging.

**Comment by 0xfe** [comment]
I merged this, but then noticed in the visual regression tests that the ratioed tuplets are broken. (They don't show ratios anymore.)

**Comment by Silverwolf90** [comment]
Looking into it.

**Pull request merged**